app/assets/javascripts/api_stub.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Created by kaceyryan on 5/2/15.
+ */
+
+var request = require('request');
+
+class API {
+  //get images in JSON object for your location
+  getImages(callback) {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        var query = "/images?lat=".concat(position.coords.latitude, "&long=", position.coords.longitude);
+        //var query = "http://localhost:3000/images?lat=34.0722&long=118.4441";
+        request(query, function (error, response, body) {
+          if (!error && response.statusCode == 200) {
+            callback(JSON.parse(body));
+          }
+        });
+      }, this.showError);
+    } else {
+      //Handle error
+      //x.innerHTML = "Geolocation is not supported by this browser.";
+    }
+  }
+
+  //post an image for your provided location
+  //Does this need a callback?
+  postImage(image, callback){
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        var query = "/images?lat=".concat(position.coords.latitude, "&long=", position.coords.longitude);
+        //var query = "http://localhost:3000/images?lat=34.0722&long=118.4441";
+        request.post({url: query, body: image}, function (error, response, body) {
+          if (!error && response.statusCode == 200) {
+            //callback(JSON.parse(body));
+            //console.log(body);
+          }
+        });
+      }, this.showError);
+    } else {
+      //Handle error
+      //x.innerHTML = "Geolocation is not supported by this browser.";
+    }
+  }
+
+  //get whether you have upvote the post yet in json object
+  getUpvote(id, callback){
+    var query = "/images/".concat(id, "/up");
+    request(query, function (error, response, body) {
+      if (!error && response.statusCode == 200) {
+        callback(JSON.parse(body));
+      }
+    });
+  }
+
+  //get whether you have downvoted the post in json
+  getDownvote(id, callback){
+    var query = "/images/".concat(id, "/down");
+    request(query, function (error, response, body) {
+      if (!error && response.statusCode == 200) {
+        callback(JSON.parse(body));
+      }
+    });
+  }
+
+  //Does this need a callback?
```

Comment:
It does--so that the React model can reconcile itself when a vote is successful. We should still update the UI instantly, though, so there isn't visible lag.